iroh-net/src/magicsock.rs

```diff
@@ -1782,7 +1794,15 @@ impl Actor {
                     inc!(Metrics, actor_tick_re_stun);
                     self.msock.re_stun("periodic");
                 }
-                Ok(()) = portmap_watcher.changed() => {
+                change = portmap_watcher.changed(), if !portmap_watcher_closed => {
+                    if change.is_err() {
+                        trace!("tick: portmap watcher closed");
+                        inc!(Metrics, actor_tick_other);
+
+                        portmap_watcher_closed = true;
```

Comment:
Is the portmapper critical?
- yes: then it should probably be restarted and a new watcher installed.
- no: then this should perhaps log on error and carry on like done here.

I'm kind of tempted to go with "no" for now, but "yes" long term.  The magicsock can work without a portmapper, just won't be as good.  But this is almost certainly a bug in our code.